### PR TITLE
Taking turns

### PR DESCRIPTION
**Pull request by JamieZafar**

### CODE REVIEW

File: src/client/game/game.js [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+export class Game extends Component {
+    componentWillMount() {
+        setSetLeaderDispatch(this.props.setLeader);
+    }
+
     render() {
         return (
             <div className="col-lg-8 text-center">
-                <h3>THIS IS GAME</h3>
+                <h3 className={style.header3}>THIS IS GAME</h3>
```
Comment (kenneth-gray) [inline src/client/game/game.js]: style.header3 is old code?

File: src/server/cloak-server.js [hunk 2]
```diff
@@ -112,7 +112,31 @@ function roomDetails(roomId, user) {
...
+    room.data.leader = leaderIndex;
     room.data.started = true;
     room.messageMembers('startGame');
+
+    setLeader(room);
     fireRoomListReload();
+    gameController(room);
+}
+
+function setLeader(room) {
```
Comment (kenneth-gray) [inline src/server/cloak-server.js]: Are there tests for this?

File: src/client/services/cloak-service.js [hunk 3]
```diff
@@ -81,3 +86,7 @@ function messageStartGame() {
 function setStartGame(dispatch) {
     startGameDispatch = dispatch;
 }
+
+function setSetLeaderDispatch(dispatch) {
```
Comment (kenneth-gray) [inline src/client/services/cloak-service.js]: Crazy name?

File: src/server/cloak-server.js [hunk 4]
```diff
@@ -112,7 +112,31 @@ function roomDetails(roomId, user) {
 
 function startGame(arg, user) {
     var room = user.getRoom();
+    var roomUsers = room.getMembers();
+    var leaderIndex = roomUsers.indexOf(user);
+    room.data.leader = leaderIndex;
```
Comment (kenneth-gray) [inline src/server/cloak-server.js]: Is leader a good name here? Should it be leaderIndex or leaderId?

Comment (kenneth-gray) [inline src/server/cloak-server.js]: Or memberIndex?

**Comment by thullSL** [comment]
:shipit:

**Comment by bumblebot1** [comment]
there is at the moment a bug when all users leave the room and the set leader crashes the server because next leader is undefined

File: src/client/game/game-reducer.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
-import { START_GAME } from './game-actions';
+import { START_GAME, SET_LEADER } from './game-actions';
 
 const initialState = {
-	started: false
+	started: false,
+	leader: 'undefined'
```
Comment (kenneth-gray) [inline src/client/game/game-reducer.js]: If a property is left off it's automatically undefined (keyword not a string).

**Comment by JamieZafar** [comment]
:ship: dat :hankey: ?

**Pull request merged**